### ISSUE TITLE
Count Bredig files

### ISSUE THREAD

**Issue by apinkney0696** [sciencehistory/scihist_digicoll#1655]
Could I please get a count of the total number of digital assets (they should all be image files) that match the following data:

Sierra Bib No.: [b1087646](https://othmerlib.sciencehistory.org/record=b1087646)
Box: 1
(aka, the number of image files made for box 1 of the Bredig collection)

This is just for reporting some data for the Bredig project grant. We have all of the other numbers we need, just missed this one. If anyone needs a small task this week, this would be helpful! Thanks in advance!

**Comment by jrochkind** [comment]
i'll do this right now

**Comment by jrochkind** [comment]
@apinkney0696  looks like 141

@eddierubeiz I figured out how to do this having the DB do the count with postgres jsonb operators... it was kinda challenging....

    Work.where(%q[json_attributes -> 'external_id' @> '[{"value": "b1087646", "category": "bib"}]'::jsonb]).
             where(%q[json_attributes -> 'physical_container' ->> 'box' = '3']).count

**Comment by eddierubeiz** [comment]
Those jsonb operators are tricky but I always have fun messing with them. Noted, and thanks for saving the recipe for posterity.

**Comment by eddierubeiz** [comment]
Re `%q`:  https://simpleror.wordpress.com/2009/03/15/q-q-w-w-x-r-s/#sq - nice to know.

**Comment by apinkney0696** [comment]
Thank you both! Hmm, this isn't the number I was expecting. There are 285 digital records created from the physical materials in Box 1 so there should be a minimum of 285 files. There is sometimes more than one page per record so I'm thinking it will likely be somewhere between 400 and 500 files. 

I looked through some of the records and I asked Jahna if there was any chance the metadata could be missing from these, and she said it's not likely. I noticed in the code above that you have 

>  where(%q[json_attributes -> 'physical_container' ->> 'box' = '3']).count

Does this need to be Box 1? 

Let me know if you need any other info from me! Thanks!!

**Comment by eddierubeiz** [comment]
Looks like a typo. I can run the same code now with 3 instead of 1.

**Comment by apinkney0696** [comment]
Thank you thank you!

**Comment by eddierubeiz** [comment]
@jrochkind - also just noting Annabel wants the asset count, not the work count.

**Comment by jrochkind** [comment]
Hm, nice catch, my example had wrong box number. Let me try again making usre I'm doing box 1...

Now I get 289 Works. 

```
Work.where(%q[json_attributes -> 'external_id' @> '[{"value": "b1087646", "category": "bib"}]'::jsonb]).
          where(%q[json_attributes -> 'physical_container' ->> 'box' = '1']).count
=> 289
```

Here is the list of all 289: https://gist.github.com/jrochkind/a8b69ef7a4c213192ad707ff7c0fbe38

OH, you want asset count, ok..... 

```
Work.where(%q[json_attributes -> 'external_id' @> '[{"value": "b1087646", "category": "bib"}]'::jsonb]).
    where(%q[json_attributes -> 'physical_container' ->> 'box' = '1']).collect(&:members).flatten.count
=> 598
````

598!

**Comment by apinkney0696** [comment]
That's it!!! Perfect-o!!
